src/useAsnyc.js

```diff
@@ -0,0 +1,47 @@
+/* eslint react/prop-types: 0 */
+// @flow
+import * as React from 'react';
+import usePromise from './utils/usePromise';
+
+class AsyncStore {
+  constructor(result, error, state, lastValue) {
+    this._result = result;
+    this._error = error;
+    this._state = state;
+    this.lastValue = lastValue;
+  }
+
+  value() {
+    return this._result || null;
```

Comment:
I think it's actually more correct to return `undefined` in these cases. Also `value` returns the latest value, so `this._result` isn't the correct field.

`return this._latestValue ? this._latestValue.inner : undefined`